sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -71,7 +73,13 @@ case class HadoopFsRelation(
 
   override def sizeInBytes: Long = {
     val compressionFactor = sqlContext.conf.fileCompressionFactor
-    (location.sizeInBytes * compressionFactor).toLong
+    val defaultSize = (location.sizeInBytes * compressionFactor).toLong
+    location match {
+      case cfi: CatalogFileIndex if sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled =>
+        CommandUtils.getSizeInBytesFallBackToHdfs(sparkSession, new Path(cfi.table.location),
```

Comment:
@shahidki31 . 
Although `fallBackToHdfsForStatsEnable=true`, this function should be invoked only when there is no available statistics. In this PR, this seems to be invoked blindly. Could you add another test case to make it sure that we don't call this function when the statistics exists. For example, after executing `analyze table t1 compute statistics;`?

Reply:
If statistics is not empty, the flow will not come to this code. Hence irrespective of `fallBackToHdfsForStatsEnable=true` or false, stats will be same. I have added a tests too, to verify the same.

https://github.com/apache/spark/blob/d943ee0a881540aa356cdce533b693baaf7c644f/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala#L42-L46

Reply:
Thank you for the explanation and the additional test case, @shahidki31 !